PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/TelemetryClientProvider.java

```diff
@@ -0,0 +1,72 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import java.io.FileNotFoundException;
+
+import java.util.Timer;
+import org.json.JSONObject;
+import java.io.InputStream;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.TimerTask;
+import java.util.stream.Collectors;
+
+/**
+ * This class reads the instrumentation key from the applicationinsights.json file
+ * and returns a TelemetryClient object with the instrumentation key set.
+ * This object is used to send telemetry data to Application Insights.
+ * The trackMemoryUsage method is used to track the memory usage of the application.
+ * The start method is used to start a timer that runs the trackMemoryUsage method every 10 seconds.
+ */
+public class TelemetryClientProvider {
```

Comment:
It would be good to do a separate PR for this feature and not mix it with the rules implementation effort.